### PR TITLE
OPRUN-3064: RPM package for Openshift-lifecycle-manager microshfit manifests 

### PR DESCRIPTION
**Pull request by jmprusi**
Depends on https://github.com/openshift/microshift/pull/2608

**Which issue(s) this PR addresses**:

Closes #<Issue Number>

### CODE REVIEW

File: packaging/rpm/microshift-olm.spec
Comment (ggiguash) [inline packaging/rpm/microshift-olm.spec:1]: Can we add this to microshift.spec file? That's the one built by ART at the moment and it contains all the RPMs

Comment (jmprusi) [reply packaging/rpm/microshift-olm.spec:1]: Done! Please, can you take a look? I have no clue about RPM specs :)

File: packaging/rpm/microshift.spec [hunk 1]
```diff
@@ -126,6 +126,13 @@ Requires: greenboot
 %description greenboot
 The microshift-greenboot package provides the Greenboot scripts used for verifying that MicroShift is up and running.
 
+%package microshift-olm-manifests
```
Comment (dhellmann) [inline packaging/rpm/microshift.spec:129]: I think we probably want this to be simply `microshift-olm`? Is there an obvious benefit to specifying "manifests"?

Comment (jmprusi) [reply packaging/rpm/microshift.spec]: Done!

File: packaging/rpm/microshift.spec [hunk 2]
```diff
@@ -127,6 +126,23 @@ Requires: greenboot
...
+Requires: microshift = %{version}
+
+%description microshift-olm-release-info
+The microshift-olm-release-info package provides release information files for this
+release. These files contain the list of container image references used by
+the Operator-lifecycle-manager and can be used to embed those images into osbuilder blueprints.
+
+%package microshift-olm
+Summary: Operator-lifecycle-manager manifests for microshift
+BuildArch: noarch
```
Comment (pmtk) [inline packaging/rpm/microshift.spec:141]: I'm wondering if we need two packages. Could `microshift-olm` contain the `release-$ARCH.json` and we'd have just one? It only contains text files so I think we could get away with one RPM

Comment (jmprusi) [reply packaging/rpm/microshift.spec]: Sounds good to me!

**Comment by pmtk** [comment]
Looks good to me. Builds and runs.
```
$ ls /usr/lib/microshift/manifests.d/001-microshift-olm 
0000_50_olm_00-catalogsources.crd.yaml           0000_50_olm_06-psm-operator.service.yaml
0000_50_olm_00-clusterserviceversions.crd.yaml   0000_50_olm_07-collect-profiles.cronjob.yaml
0000_50_olm_00-installplans.crd.yaml             0000_50_olm_07-olm-operator.deployment.yaml
0000_50_olm_00-namespace.yaml                    0000_50_olm_08-catalog-operator.deployment.yaml
0000_50_olm_00-olmconfigs.crd.yaml               0000_50_olm_09-aggregated.clusterrole.yaml
0000_50_olm_00-operatorconditions.crd.yaml       0000_50_olm_11-olm-operators.configmap.removed.yaml
0000_50_olm_00-operatorgroups.crd.yaml           0000_50_olm_12-olm-operators.catalogsource.removed.yaml
0000_50_olm_00-operators.crd.yaml                0000_50_olm_13-operatorgroup-default.yaml
0000_50_olm_00-packageserver.pdb.yaml            0000_50_olm_14-packageserver.subscription.removed.yaml
0000_50_olm_00-pprof-config.yaml                 0000_50_olm_15-csv-viewer.rbac.yaml
0000_50_olm_00-pprof-rbac.yaml                   0000_50_olm_99-operatorstatus.yaml
0000_50_olm_00-pprof-secret.yaml                 0000_90_olm_00-service-monitor.yaml
0000_50_olm_00-subscriptions.crd.yaml            0000_90_olm_01-prometheus-rule.yaml
0000_50_olm_01-olm-operator.serviceaccount.yaml  image-references
0000_50_olm_02-olmconfig.yaml                    kustomization.yaml
0000_50_olm_02-services.yaml                     release-aarch64.json
0000_50_olm_06-psm-operator.deployment.yaml      release-x86_64.json
0000_50_olm_06-psm-operator.servicemonitor.yaml
```

```
$ oc get pods -A
NAMESPACE                              NAME                                       READY   STATUS    RESTARTS       AGE
kube-system                            csi-snapshot-controller-6749df765b-2xjsx   1/1     Running   0              3m52s
kube-system                            csi-snapshot-webhook-dbc4f64f4-7fmgm       1/1     Running   0              3m49s
openshift-dns                          dns-default-5lr5b                          2/2     Running   0              3m28s
openshift-dns                          node-resolver-b86nb                        1/1     Running   0              3m52s
openshift-ingress                      router-default-5758bf4d7c-bkzbf            1/1     Running   0              3m49s
openshift-operator-lifecycle-manager   catalog-operator-7f69d8cdc9-24vlx          1/1     Running   0              96s
openshift-operator-lifecycle-manager   olm-operator-5787c5ff8f-nh5tz              1/1     Running   0              96s
openshift-ovn-kubernetes               ovnkube-master-gjgql                       4/4     Running   0              3m52s
openshift-ovn-kubernetes               ovnkube-node-wctl8                         1/1     Running   0              3m51s
openshift-service-ca                   service-ca-5595f997cb-95bk4                1/1     Running   0              3m49s
openshift-storage                      topolvm-controller-75f469bbc6-55npv        5/5     Running   3 (100s ago)   3m52s
openshift-storage                      topolvm-node-2sm69                         4/4     Running   0              3m28s
```

File: packaging/rpm/microshift.spec [hunk 3]
```diff
@@ -346,6 +368,9 @@ systemctl enable --now --quiet openvswitch || true
 %{_sysconfdir}/greenboot/red.d/40_microshift_pre_rollback.sh
 %{_datadir}/microshift/functions/greenboot.sh
 
+%files olm
+%{_prefix}/lib/microshift/manifests.d/001-microshift-olm/*
+
 # Use Git command to generate the log and replace the VERSION string
 # LANG=C git log --date="format:%a %b %d %Y" --pretty="tformat:* %cd %an <%ae> VERSION%n- %s%n" packaging/rpm/microshift.spec
 %changelog
```
Comment (ggiguash) [inline packaging/rpm/microshift.spec:376]: Need to add a change log

File: packaging/rpm/microshift.spec [hunk 4]
```diff
@@ -127,6 +126,13 @@ Requires: greenboot
 %description greenboot
 The microshift-greenboot package provides the Greenboot scripts used for verifying that MicroShift is up and running.
 
+%package olm
+Summary: Operator-lifecycle-manager manifests for microshift
```
Comment (ggiguash) [inline packaging/rpm/microshift.spec:130]:
```suggestion
Summary: Operator Llifecycle Manager xomponents for MicroShift
```

File: packaging/rpm/microshift.spec [hunk 5]
```diff
@@ -346,6 +368,9 @@ systemctl enable --now --quiet openvswitch || true
 %{_sysconfdir}/greenboot/red.d/40_microshift_pre_rollback.sh
 %{_datadir}/microshift/functions/greenboot.sh
 
+%files olm
```
Comment (ggiguash) [inline packaging/rpm/microshift.spec:371]: Also need this here to register a directory
```
%dir %{_prefix}/lib/microshift/manifests.d/001-microshift-olm
```

Comment (jmprusi) [reply packaging/rpm/microshift.spec:371]: done, PTAL

File: packaging/rpm/microshift.spec (hunk 4 above)
Comment (jmprusi) [reply packaging/rpm/microshift.spec]: done, PTAL

File: packaging/rpm/microshift.spec [hunk 6]
```diff
@@ -127,6 +126,13 @@ Requires: greenboot
 %description greenboot
 The microshift-greenboot package provides the Greenboot scripts used for verifying that MicroShift is up and running.
 
+%package olm
+Summary: Operator Llifecycle Manager components for MicroShift
```
Comment (ggiguash) [inline packaging/rpm/microshift.spec:130]:
```suggestion
Summary: Operator Lifecycle Manager components for MicroShift
```

Comment (jmprusi) [reply packaging/rpm/microshift.spec]: fixed

**Pull request merged**